### PR TITLE
Fixing developer build on clang >=15

### LINKED ISSUE #163: MPS fails developer build on clang 15
I was wondering how to get more coverage of arm64 and realised I had [Termux](https://termux.dev/en/) installed on my phone.

Dataflow analysis in the hot build of mps.c throws a couple of warnings for this particular version of Clang.

```
~/git/mps/code $ uname -a
Linux localhost 4.9.218+ #1 SMP PREEMPT Mon Dec 7 20:15:05 UTC 2020 aarch64 Android
~/git/mps/code $ cc -dumpmachine
aarch64-unknown-linux-android24
~/git/mps/code $ cc --version
clang version 15.0.7
Target: aarch64-unknown-linux-android24
Thread model: posix
InstalledDir: /data/data/com.termux/files/usr/bin
~/git/mps/code $ make -f lia6ll.gmk
make[1]: Entering directory '/storage/emulated/0/git/mps/code'
make[2]: Entering directory '/storage/emulated/0/git/mps/code'
lia6ll: lia6ll/hot/mps.o
In file included from mps.c:88:
./poolawl.c:1060:9: error: variable 'preservedInPlaceCount' set but not used
      [-Werror,-Wunused-but-set-variable]
  Count preservedInPlaceCount = (Count)0;
        ^
In file included from mps.c:89:
./poollo.c:316:9: error: variable 'preservedInPlaceCount' set but not used
      [-Werror,-Wunused-but-set-variable]
  Count preservedInPlaceCount = (Count)0;
        ^
2 errors generated.
make[2]: *** [comm.gmk:615: lia6ll/hot/mps.o] Error 1
make[2]: Leaving directory '/storage/emulated/0/git/mps/code'
make[1]: *** [comm.gmk:395: target] Error 2
make[1]: Leaving directory '/storage/emulated/0/git/mps/code'
make: *** [comm.gmk:366: mps.a] Error 2
```

The deployment build `cc -c mps.c` is fine.

### PR DESCRIPTION
**Pull request by rptb1**
Fixes #163 .

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
These warnings break the WebAssembly (#175) build as well since that also uses a newer version of clang.

**Comment by rptb1** [comment]
I can confirm that this branch is required for clang-15 and clang-16 and clang-17.  Can be checked with e.g. `make -C code -f lii6ll.gmk CC=clang-15 VARIETY=hot clean lii6ll/hot/mps.o`

**Comment by rptb1** [comment]
This branch is required by #145 . See https://github.com/Ravenbrook/mps/pull/145/checks?check_run_id=11847545950 .

**Comment by rptb1** [comment]
@waywardmonkeys says in keybase://chat/rptb1,bmitchener/530 :
> Great, 164 is most annoying as I have to keep modifying things to let it build :)

File: code/amcss.c [hunk 1]
```diff
@@ -95,7 +95,7 @@ static void report(void)
 
 static mps_addr_t make(size_t rootsCount)
 {
-  static unsigned long calls = 0;
+  static volatile unsigned long calls = 0; /* for use from debugger */
```
Comment (rptb1) [inline code/amcss.c:98]: m. Could refer to https://github.com/Ravenbrook/mps/commit/0c5852a0838590ea2815efa599a3e5b4ae2d5b5c and/or explain that this is to help with watchpoints.

File: code/poolawl.c [hunk 2]
```diff
@@ -1057,7 +1057,7 @@ static void awlSegReclaim(Seg seg, Trace trace)
   Bool hasBuffer = SegBuffer(&buffer, seg);
   Format format = pool->format;
   Count reclaimedGrains = (Count)0;
-  Count preservedInPlaceCount = (Count)0;
+  STATISTIC_DECL(Count preservedInPlaceCount = (Count)0)
```
Comment (rptb1) [inline code/poolawl.c:1060]: q. Do these STATISTIC macros get compiled in (and therefore checked) for any of our varieties?

Comment (rptb1) [reply code/poolawl.c:1060]: Answer: yes, in COOL. https://github.com/Ravenbrook/mps/blob/4f5f119b23c92212f2647de383e13aee07bb2c1b/code/config.h#L50-L54

**Review by rptb1 — COMMENTED** [review]
Executing [proc.review.check](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#54-review-checking)

1. Start time 11:12
2. Checked compilation with clang-15 with `make -C code -f lii6ll.gmk CC=clang-15 VARIETY=hot clean testci`.  All passed.

**Review by UNAA008 — COMMENTED** [review]
I have reviewed the changed files in the context of Issue #163 and am satisfied that the changes are correct.
There would not have been time to review the whole files in an express review.
No defects found.

**Comment by rptb1** [comment]
Executing [proc.review.express](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#12-express-review)

1. Start time 10:47
2. We'll be applying [rule.generic](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/rule.generic.rst) and [rule.code](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/rule.code.rst)
3. Im. entry.universal shouldn't demand reviews of issues.
4. Called in @UNAA008 
5. Paused 10:50 while waiting for @UNAA008 
6. Resume 11:07.
7. Checking started 11:12 for 10 mins.
8. Logging meeting started at 11:22.
9. Im. @UNAA008 asks that the [interdiction on checking using diffs](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#542-checking) makes it clear that you're not expected to check *all* of the file *as changed*.
10. @UNAA008 found nothing. @rptb1 found 1 minor defect and 1 question.
11. proc.review.express.brainstorm: @rptb1 I could've used `git blame` when I was making the change to add an appropriate comment.
12. Logging ends 11:30.
13. Estimated defects remaning: 0
14. Revised change passed.
15. review.exit.calc:
  - hours used: 1
  - hours saved: not a lot
  - major defect remaining: 0

**Comment by rptb1** [comment]
Executing [proc.merge.pull-request](https://github.com/Ravenbrook/mps/blob/e697cebcb9a7c0a8e4a0aac496137b4840622fdd/procedure/pull-request-merge.rst)

1. Start time 11:37
2. The compiler is the automated test for this.
3. Waiting for CI to pass until 11:50
4. Merge took 7 minutes.

**Pull request merged**